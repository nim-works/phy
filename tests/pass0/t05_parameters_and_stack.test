discard """
  description: "
    Ensure that a procedure with both parameters and a frame pointer
    works
  "
  output: "(Done 2)"
"""
(TypeDefs
  (ProcTy (Int 8) (Int 8) (Int 8))
  (ProcTy (Int 8)))
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 0) 8
    (Locals (Int 8) (Int 8) (Int 8))
    (List
      (Block
        (Params (Local 0) (Local 1) (Local 2))
        (Store (Int 8)
          (Copy (Local 2))
          (Copy (Local 0)))
        (Return
          (Copy (Local 1))))))

  (ProcDef (Type 1) 0 (Locals)
    (List
      (Block (Params)
        (Return
          (Call (Proc 0) (IntVal 1) (IntVal 2)))))))
