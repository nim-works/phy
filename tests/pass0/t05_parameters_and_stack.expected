.type t0 (int, int) -> int
.type t1 () -> int
.start t0 p0
.local lo0 int
.local lo1 int
.local lo2 int
  StackAlloc 8
  PopLocal lo2
  PopLocal lo1
  PopLocal lo0
  GetLocal lo2
  GetLocal lo0
  WrInt64 0
  GetLocal lo1
  Ret
.end
.start t1 p1
  LdImmInt 1
  LdImmInt 2
  Call p0 2
  Ret
.end
